.github/workflows/ci-emscripten.yml

```diff
@@ -0,0 +1,155 @@
+      NODE_VERSION: 20
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Set up Emscripten toolchain
+        uses: mymindstorm/setup-emsdk@6ab9eb1bda2574c4ddb79809fc9247783eaf9021 # v14
+        with:
+          version: ${{ env.EMSCRIPTEN_VERSION }}
+          actions-cache-folder: emsdk-cache
+
+      - name: Set up Node.js
+        uses: actions/setup-node@1d0ff469b7ec7b3cb9d8673fde0c81c44821de2a # v4.2.0
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install pyodide-build
+        run: pip install pyodide-build
+
+      - name: Restore WASM library directory from cache
+        id: cache-wasm-library-dir
+        uses: actions/cache/restore@d4323d4df104b026a6aa633fdb11d772146be0bf # v4.2.2
+        with:
+          path: ${{ github.workspace }}/wasm-library-dir
+          key: wasm-library-dir-${{ hashFiles('.github/workflows/ci-emscripten.yml') }}
+
+      - name: Build libgmp
+        if: steps.cache-wasm-library-dir.outputs.cache-hit != 'true'
+        env:
+          CFLAGS: "-fPIC"
+          WASM_LIBRARY_DIR: ${{ github.workspace }}/wasm-library-dir
+        run: |
+          curl -L https://ftp.gnu.org/gnu/gmp/gmp-6.3.0.tar.xz -o gmp-6.3.0.tar.xz
+          tar -xf gmp-6.3.0.tar.xz
+
+          cd gmp-6.3.0
+
+          emconfigure ./configure \
+            --disable-dependency-tracking \
+            --host none \
+            --disable-shared \
+            --enable-static \
+            --enable-cxx \
+            --prefix=${{ env.WASM_LIBRARY_DIR }}
+          emmake make -j $(nproc)
+          emmake make install
+
+      - name: Build libmpfr
+        if: steps.cache-wasm-library-dir.outputs.cache-hit != 'true'
+        env:
+          CFLAGS: "-fPIC"
+          WASM_LIBRARY_DIR: ${{ github.workspace }}/wasm-library-dir
+        run: |
+          curl -L https://ftp.gnu.org/gnu/mpfr/mpfr-4.2.1.tar.xz -o mpfr-4.2.1.tar.xz
+          tar -xf mpfr-4.2.1.tar.xz
+
+          cd mpfr-4.2.1
+
+          emconfigure ./configure \
+            --disable-dependency-tracking \
+            --disable-shared \
+            --with-gmp="${{ env.WASM_LIBRARY_DIR }}" \
+            --prefix=${{ env.WASM_LIBRARY_DIR }}
+          emmake make -j $(nproc)
+          emmake make install
+
+      # might need patch
+      - name: Build flint
+        if: steps.cache-wasm-library-dir.outputs.cache-hit != 'true'
+        env:
+          CFLAGS: "-fPIC"
+          WASM_LIBRARY_DIR: ${{ github.workspace }}/wasm-library-dir
+        run: |
+          curl -L https://github.com/flintlib/flint/releases/download/v3.2.0-rc1/flint-3.2.0-rc1.tar.xz -o flint-3.2.0-rc1.tar.xz
+          tar -xf flint-3.2.0-rc1.tar.xz
+
+          cd flint-3.2.0-rc1
```

Comment:
We can checkout FLINT main here,

Reply:
Addressed in 21f6560e1b694f08388747ff2a31428b296fd411